src/drake/core.clj

```diff
@@ -106,8 +106,13 @@
    it. This is safe to do since it's the default filesystem,
    but it gives us a bit easier compatibility with existing tools."
   [filename]
-  (let [n (dfs/normalized-path filename)]
```

Comment:
@chen-factual What do you think about a function that wraps functions so that they correctly handle optional inputs? Like instead of making `normalize-filename-for-run` and similar functions call `parser/modify-filename` themselves, you could just rename the old definition to `normalize-filename-for-run*`, and then `(def normalize-filename-for-run (wrap-optional-handling normalize-filename-for-run*))` or whatever.